flow-typed/custom/superagent.js

```diff
@@ -0,0 +1,307 @@
+    mkcol(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    move(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    notify(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    options(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    patch(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    post(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    propfind(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    proppatch(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    purge(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    put(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    report(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    saveCookies(res: superagent$Response): void;
+    search(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    subscribe(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    trace(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    unlock(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+    unsubscribe(
+        url: string,
+        callback?: superagent$CallbackHandler,
+    ): superagent$SuperAgentRequest;
+}
+
+declare interface superagent$HTTPError extends Error {
+    status: number;
+    text: string;
+    method: string;
+    path: string;
+}
+
+declare interface superagent$Response extends NodeJS.ReadableStream {
+    accepted: boolean;
+    badRequest: boolean;
+    body: any;
+    charset: string;
+    clientError: boolean;
+    error: false | superagent$HTTPError;
+    files: any;
+    forbidden: boolean;
+    get(header: string): string;
+    get(header: "Set-Cookie"): string[];
+    header: any;
+    info: boolean;
+    links: object;
+    noContent: boolean;
+    notAcceptable: boolean;
+    notFound: boolean;
+    ok: boolean;
+    redirect: boolean;
+    serverError: boolean;
+    status: number;
+    statusType: number;
+    text: string;
+    type: string;
+    unauthorized: boolean;
+    xhr: XMLHttpRequest;
+    redirects: string[];
+}
+
+declare interface superagent$Request extends Promise<superagent$Response> {
+    cookies: string;
+    method: string;
+    url: string;
```

Comment:
These were on the SuperAgentRequest type in the typescript code, but they are clearly added to all requests and we want them here, so I moved them.